src/Storefront/Resources/app/storefront/src/plugin/google-analytics/events/view-item.event.js

```diff
@@ -12,12 +12,26 @@ export default class ViewItemEvent extends AnalyticsEvent
             return;
         }
 
-        const form = DomAccessHelper.querySelector(document, '#productDetailPageBuyProductForm');
-        const productId = this.findProductId(DomAccessHelper.querySelectorAll(form, 'input'));
-        const productName = DomAccessHelper.querySelector(form, 'input[name=product-name]').value;
+        const productItemElement = DomAccessHelper.querySelector(document, '[itemtype="https://schema.org/Product"]', false);
```

Comment:
Hi @aragon999, 

thanks for your contribution. 👍 

So basically you switched to those selectors because they are not removed when the product is sold-out, unlike the form selectors? And the overall goal ist that the tracking is also triggered even when the product is sold-out?

Are there actual cases where e.g. `productItemElement` are not there, or was this implemented to be extra safe?

Reply:
hey @tobiasberge,

thank you for the review.

So in general this should not happen, but I remembered one customer where we did something wrong (when changing the templates in combinations with upstream changes), such that this element was missing on the page.

So my thoughts were, that it would be nice as a template developer to get direct feedback what exactly is wrong. Especially since debugging Google Analytics is annoying most of the times.

I can also remove this extra check.

Reply:
Hi, 

thank you for clarifying. 

I would also prefer something with a bit of error handling to support the developer. But I'm not sure if it will work for `productIdElement` and `productNameElement` as expected because `DomAccessHelper.querySelector` is still in default scrict mode here. I assume it will throw the hard error when the element is not present:

https://github.com/shopware/platform/blob/trunk/src/Storefront/Resources/app/storefront/src/helper/dom-access.helper.js#L116

Maybe it would also be sufficient to not have the extra check and leave the `DomAccessHelper.querySelector` in strict mode because then it will throw an error with a similar message that the selector does not exist.

Reply:
Oh, I actually meant them to be also in non strict mode and rather do not throw a hard error such that other things continue to work (maybe one uses a custom event, which is than also not triggered).

I would suggest to remove the strict mode for the `productIdElement` and `productNameElement`, but if desired, I can also keep the strict mode and remove the warning.

Reply:
I just updated it according to my suggestion, if you like to have that changed, please let me know :-) in a new comment.

Reply:
@aragon999 Hi, sorry for my late reply. I think this looks all right, thank you for the effort. 👍  We will now import the pull request.